core/ecschema-metadata/src/Metadata/OverrideFormat.ts

```diff
@@ -108,7 +110,35 @@ export class OverrideFormat {
   public static isOverrideFormat(object: any): object is OverrideFormat {
     const overrideFormat = object as OverrideFormat;
 
-    return overrideFormat !== undefined && overrideFormat.name !== undefined && overrideFormat.parent !== undefined &&
-             overrideFormat.parent.schemaItemType === SchemaItemType.Format;
+    return overrideFormat !== undefined && overrideFormat.name !== undefined && overrideFormat.parent !== undefined && overrideFormat.parent.schemaItemType === SchemaItemType.Format;
+  }
+
+  /**
+   * Returns a JSON object that contains the specification for OverrideFormat.
+   */
+  public toJSON(): SchemaItemFormatProps {
+    const formatJson = this.parent.toJSON() as Mutable<SchemaItemFormatProps>;
```

Comment:
I don't know that a Format Override has the same props as a format that is a schema item.  With this implementation it appears that you will return format json that is identical to the parent format just with the overridden values changed.  How is someone supposed to determine if this json is the original format or an override of the format?

One way to implement this would be to serialize the name of the parent format and the overridden values.  This would match how we actually serialize the value in xml and json.  Maybe this is not what @grigasp wants.  Another option is to serialize the entire set of format props but not serialize the schema item portions.  Then add a parent property with the parent formats name.  This would let us identify the json as a format override, it would not let us directly determine which attributes were overridden but it could be determined by comparing with the original format.

I also think the documentation here needs to be updated to be more explicit what it is doing.  The 'specification for the OverrideFormat' is really just the parent name plus the overrides.  Or well at least it is to me but I could see someone having a different opinion.  So we should better describe what the method is doing and how the result should be used.  Normally toJSON() methods are used for serialization, this would not be.

Reply:
We want to serialize `Format` and `OverrideFormat` into [`core-quantity` FomatProps](https://github.com/iTwin/itwinjs-core/blob/cc29d0d9fba7dbf11b2a87e3b97c8bf80f21c105/core/quantity/src/Formatter/Interfaces.ts#L15). At the moment we need to check what type of format we get from Schema and [convert accordingly](https://github.com/iTwin/itwinjs-core/blob/cc29d0d9fba7dbf11b2a87e3b97c8bf80f21c105/presentation/common/src/presentation-common/KoqPropertyValueFormatter.ts#L128).

From your comment it looks like that this should not be done using `toJSON` method? If that's the case I think a separate function that converts `Fomat` and `OverrideFormat` into `core-quatinty` `FormatProps` should be to introduced and `toJSON` can be implemented in a way that makes the most sense for Schema serialization/deserialization.